src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AncientAltarListener.java

```diff
@@ -156,8 +156,18 @@ private void usePedestal(@Nonnull Block pedestal, @Nonnull Player p) {
             Slimefun.runSync(() -> removedItems.remove(uuid), 30L);
 
             entity.remove();
-            p.getInventory().addItem(pedestalItem.getOriginalItemStack(entity));
             p.playSound(pedestal.getLocation(), Sound.ENTITY_ITEM_PICKUP, 1F, 1F);
+
+            /*
+             * Fixes #3476
+             * Drop the item instead if the player's inventory is full and
+             * no stack space left else add the item in the inventory
+             */
+            for (Map.Entry<Integer, ItemStack> stackEntry : p.getInventory().addItem(pedestalItem.getOriginalItemStack(entity)).entrySet()) {
```

Comment:
I think declaring the map separately from the for loop here would be far easier to read and far neater.